### ISSUE TITLE
GZDoom language lump?

### ISSUE THREAD

**Issue by mc776** [freedoom/freedoom#1128]
Something to consider down the line to allow in-game messages to be displayed in multiple languages.

Leaving this here in case someone out there's got a mind to actually do the translation but is deterred by the lack of support.

**Comment by RicardoLuis0** [comment]
> deterred by the lack of support
just wanted to mention here, but language lump will now work https://github.com/ZDoom/gzdoom/issues/2681 https://github.com/ZDoom/gzdoom/pull/2682

**Comment by aeonlight** [comment]
bumping this, as I'm currently failing to overwrite language information with a LANGUAGE.csv file in my own mod (running UZDoom release ver 4.14.3, FREEDOOM2.WAD release ver 0.13.0)

**Comment by mc776** [comment]
Does it work with id Doom? Could be an issue with LANGUAGE versus DEHACKED.

**Comment by aeonlight** [comment]
it does work properly with id doom, yes. if you'd like, I could grab some screenshots of the text in-game? but it does seem like a LANGUAGE vs DEHACKED issue, I agree.

**Comment by mc776** [comment]
I just remembered FD has UMAPINFO.

Does the LANGUAGE lump work if UMAPINFO is removed?

**Comment by aeonlight** [comment]
odd. I've opened up my copy of FREEDOOM2.WAD in slade, and I can't find _any_ mapinfo lumps, _or_ a language lump. there is a dehacked lump and a dmxgusc lump, though. (I did get this wad from the download link on freedoom.github.io, so it shouldn't be modified.)

so no, it doesn't work if umapinfo is removed, since that's not in here to begin with.

**Comment by mc776** [comment]
Did you check for a **U**MAPINFO lump? Or is this an older version of FD?

**Comment by aeonlight** [comment]
yes, I checked for umapinfo. I checked every lump in the wad. and I've already listed the version above - 0.13.0, the latest release. there is no umapinfo, there is no mapinfo of any type. that's why I wrote "any mapinfo lumps" instead of "the mapinfo lump".

**Comment by Anonysussy** [comment]
Stable release doesn't count for bug reports, it's the nightly builds that does.

**Comment by aeonlight** [comment]
... what?


why?


in what world does "you can't report bugs for the stable release" make sense?? this is an issue with the release available through the download link that most people are going to use.
but sure, fine, I'll download a nightly build if you can show me where to grab it, and see if I have the same problem.

**Comment by mc776** [comment]
Just wanted to be 110% sure we were talking about the same lump, because I don't remember if the latest FD stable had UMAPINFO at all and between all these different info things it gets really confusing.

Anyway I think this is best continued in [that UZDoom issue I posted](https://github.com/UZDoom/UZDoom/issues/1332) because this does seem more a UZDoom bug than a Freedoom one.

**Comment by aeonlight** [comment]
I thought it might be, but wanted confirmation first - I'll move over there, then. thanks for the help

**Comment by Anonysussy** [comment]
> ... what?
> 
> why?
> 
> in what world does "you can't report bugs for the stable release" make sense?? this is an issue with the release available through the download link that most people are going to use. but sure, fine, I'll download a nightly build if you can show me where to grab it, and see if I have the same problem.

That's because if a bug exist in the stable release, but fixed in nightly, this could result in some false positives, and the stable release had been out for a while now. 

Also about the nightly builds, it's in the bottom part of the download page as "Development snapshots" and I did ask fraggle at some point to change it, but he insisted to keep it that way because he wanted to prioritize the stable release as the main version for regular playthrough. 

Autobuilds link for convenience:
http://freedoom.soulsphere.org/

**Comment by Anonysussy** [comment]
And I should specify because my comment caught you off-guard, but more people bug reported using the nightly builds because it's more up to date and better guarantee for testing if the stable release didn't line up with the new changes.

**Comment by aeonlight** [comment]
> That's because if a bug exist in the stable release, but fixed in nightly, this could result in some false positives, and the stable release had been out for a while now.

wouldn't it make more sense to tell people to make sure their issue isn't a duplicate before making an issue, and close issues solved in nightly as duplicates when someone does report it from stable? instead of disallowing reports from stable entirely, like you initially stated is the case??

speaking of, there's no clear issue submission rules to begin with. the popup link at the top of the page doesn't go to guidelines on filing issues, in general or for this specific project. this whole thing is confusing and unintuitive from top to bottom, unless you come in _already_ knowing what you're doing.

> Also about the nightly builds, it's in the bottom part of the download page as "Development snapshots" and I did ask fraggle at some point to change it, but he insisted to keep it that way because he wanted to prioritize the stable release as the main version for regular playthrough.
> 
> Autobuilds link for convenience: http://freedoom.soulsphere.org/

thanks. I've confirmed my language replacement IS working on the nightly build, which includes the aforementioned UMAPINFO and a LANGUAGE lump, neither of which were in my copy of the stable release, so this issue does appear to be solved for freedoom's purposes.
I am keeping the UZDOOM issue open though, as this probably shouldn't have happened in the first place.

**Comment by Anonysussy** [comment]
> this whole thing is confusing and unintuitive from top to bottom, unless you come in _already_ knowing what you're doing.

Ig an unrealistic expectations of mine, but it works with other GitHub projects once you figured out how it works.
